Parser/pegen/pegen.c

```diff
@@ -911,6 +911,46 @@ _PyPegen_number_token(Parser *p)
                     p->arena);
 }
 
+static int // bool
+newline_in_string(Parser *p, const char *cur)
+{
+    for (char c = *cur; cur >= p->tok->buf; c = *--cur) {
+        if (c == '\'' || c == '"') {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+static int // bool
+bad_single_statement(Parser *p)
+{
+    const char *cur = strchr(p->tok->buf, '\n');
+
+    /* Newlines are allowed if preceded by a line continuation character
+       or if they appear inside a string. */
+    if (!cur || *(cur - 1) == '\\' || newline_in_string(p, cur)) {
```

Comment:
Could we use `strpbrk` instead of `newline_in_string` ?

Reply:
Not sure. We would need to create a new buffer that would only hold the content of `p->tok->buf` up until `cur`, right? We only want to see if there is a quote between those two.